src/languageSupport/languages/sql/connection.ts

```diff
@@ -0,0 +1,240 @@
+    }
+
+    return changeInBlock || deletionFromBlock
+  }
+
+  _setCompositionHandlers() {
+    this._model.onDidChangeContent(e => {
+      const shouldEndSync = e.changes.some(
+        change =>
+          this._editorChangeIsWithinComposition(change) &&
+          !this._couldBeFromComposition(change)
+      )
+      if (shouldEndSync) {
+        // use setTimeout to remove race condition
+        // have changes propogation first to SqlEditorMonaco.onChange()
+        setTimeout(() => {
+          this._callbackSetSession({
+            composition: {synced: false},
+          })
+          this._dispatcher(notify(compositionEnded()))
+        }, 0)
+      }
+    })
+  }
+
+  _buildComposition(
+    _toAdd: Partial<CompositionSelection>,
+    _toRemove: Partial<CompositionSelection> = null
+  ) {
+    let composition = ['SELECT *']
+    let lines = 1
+    if (this._session.measurement) {
+      composition.push(`FROM "${this._session.measurement}"`)
+      lines++
+    }
+
+    const fieldsExpr = this._session.fields
+      .map(f => `"${f}" IS NOT NULL`)
+      .join(' OR ')
+    const tagValuesExpr = Object.entries(
+      groupedTagValues(this._session.tagValues)
+    )
+      .map(
+        ([key, values]) =>
+          `"${key}" IN (${values.map(v => castType(v)).join(',')})`
+      )
+      .join(' AND ')
+
+    const timeRangeExpr = buildTimeRange(this._timeRange)
+
+    switch (
+      `${this._session.fields.length > 0}|${this._session.tagValues.length > 0}`
+    ) {
+      case `true|true`:
+        composition = composition.concat([
+          'WHERE',
+          timeRangeExpr,
+          'AND',
+          `(${fieldsExpr})`,
+          'AND',
+          tagValuesExpr,
+        ])
+        lines += 6
+        break
+      case `true|false`:
+        composition = composition.concat([
+          'WHERE',
+          timeRangeExpr,
+          'AND',
+          `(${fieldsExpr})`,
+        ])
+        lines += 4
+        break
+      case `false|true`:
+        composition = composition.concat([
+          'WHERE',
+          timeRangeExpr,
+          'AND',
+          tagValuesExpr,
+        ])
+        lines += 4
+        break
+      default:
+        composition = composition.concat(['WHERE', timeRangeExpr])
+        lines += 2
+        break
+    }
+    return {
+      composition: composition.join('\n'),
+      lines,
+      lenLastLine: composition[composition.length - 1].length,
+    }
+  }
+
+  _updateComposition(
+    toAdd: Partial<CompositionSelection>,
+    toRemove: Partial<CompositionSelection>
+  ) {
+    const {composition, lines, lenLastLine} = this._buildComposition(
+      toAdd,
+      toRemove
+    )
+
+    // replacement Range
+    const startLineNumber = this._compositionRange?.start?.line ?? 1
+    const endLineNumber = this._compositionRange?.end?.line ?? 1
+    const addNewLine = startLineNumber == 1 && endLineNumber == 1
+    const endColumn = addNewLine ? 1 : Infinity
```

Comment:
Same suggestion as above for boolean names

`shouldAddNewLine`
`shouldEndColumn`

Reply:
Good idea on the `shouldAddNewLine`, as it is a bool.
The `endColumn` is a number, representing the `endColumn` used in the monaco-editor range object (a few lines below).